main.py

```diff
@@ -465,38 +468,47 @@ def performEvaluation(issue_data) -> Result:
 
         jdk_name = f"amazon-corretto-{version}"
         jdk_tar_name = f"{jdk_name}.tar.gz"
-        jdk_abs_path = os.path.abspath(jdk_tar_name)
-        if not os.path.exists(jdk_abs_path):
-            download_with_wget(jdk_url, jdk_tar_name)
-        
-        extracted_jdk_name = jdk_name + "."+ "jdk"
-        extracted_jdk_abs_path = os.path.abspath(extracted_jdk_name)
-        if os.path.exists(extracted_jdk_abs_path):
-            shutil.rmtree(extracted_jdk_abs_path)
-        if os.path.exists(jdk_abs_path):
-            extract_tar_gz(jdk_abs_path)
+        jdk_tar_abs_path = os.path.abspath(jdk_tar_name)
+
+        if not os.path.exists(jdk_tar_abs_path):
+            download_with_wget(jdk_url, jdk_tar_abs_path)
         
-        set_directory_exec_permission(extracted_jdk_abs_path)
+        extracted_jdk_name = ""
+        if op == "linux":
+            extracted_jdk_abs_path = os.path.abspath(jdk_name) #/../amazon-corretto-8
+        else:
+            extracted_jdk_abs_path = os.path.abspath(jdk_name) + ".jdk" #//.//amazon-corretto-8.jdk
 
-        jdk_home = os.path.join(extracted_jdk_abs_path, "Contents", "Home")
-        os.environ["JAVA_HOME"] = jdk_home 
+        if not os.path.exists(extracted_jdk_abs_path):
+            os.makedirs(extracted_jdk_abs_path, exist_ok=True)
+            if os.path.exists(extracted_jdk_abs_path):
+                extract_and_rename(jdk_tar_abs_path, extracted_jdk_abs_path)
+        
+        if op == "linux":
+            java_path = os.path.join(extracted_jdk_abs_path, "bin", "java")
```

Comment:
Why `java` and not `javac`? We're trying to run a compilation pass, so it makes more sense to run `javac` than to run `java`.

Reply:
I am following this from CF document https://checkerframework.org/manual/#external-tools
 
> Option 3: Whenever this document tells you to run javac, instead run checker.jar via java (not javac) as in:
>   java -jar "$CHECKERFRAMEWORK/checker/dist/checker.jar" -cp "myclasspath" -processor nullness MyFile.java

I need to invoke the `javac` from jdk. When i use` /jdk/bin/javac -cp checker.jar ..`..  i get error that annotation "Annotation processor 'guieffect' not found" error. 
I don't know how to use `javac` and which `flags and jar` to pass other than above one.

Reply:
Thanks for this citation. Please add a comment explaining this to the code.